src/cli/api/trackingplans.ts

```diff
@@ -121,15 +124,31 @@ export function parseTrackingPlanName(name: string): { id: string; workspaceSlug
   return {
     id,
     workspaceSlug,
+    APIVersion: 'v1',
   };
 }
 
-export function toTrackingPlanURL(name: string): string {
-  const { id } = parseTrackingPlanName(name);
-  return `https://api.rudderstack.com/trackingplans/${id}`;
+export function toTrackingPlanURL(trackingPlan: RudderAPI.TrackingPlan): string {
+  if (!trackingPlan.creationType) {
+    const { id } = parseTrackingPlanName(trackingPlan.name);
+    return `https://api.rudderstack.com/trackingplans/${id}`;
+  }
+  return `https://api.rudderstack.com/tracking-plans/${trackingPlan.id}`;
+}
+
+export function toTrackingPlanId(trackingPlan: RudderAPI.TrackingPlan): string {
+  if (!trackingPlan.creationType) {
+    const { id } = parseTrackingPlanName(trackingPlan.name);
+    return id;
+  }
+  return trackingPlan.id;
 }
 
-export function toTrackingPlanId(name: string): string {
-  const { id } = parseTrackingPlanName(name);
-  return id;
+export function getTrackingPlanName(trackingPlan: RudderAPI.TrackingPlan): string {
+  // throw new Error(`Unable to parse Tracking Plan name: ${JSON.stringify(trackingPlan)}`);
+  if (!trackingPlan.creationType) {
+    return trackingPlan.display_name;
+  }
+
+  return trackingPlan.name;
```

Comment:
```suggestion
export function getTrackingPlanName(name, display_name, creationType): string {
  return !!creationType ? trackingPlan.name : trackingPlan.display_name;
```
old tracking plan object is a heavy object. consider sending dto or params containing only necessary ones